optimade/validator/validator.py

```diff
@@ -827,6 +827,15 @@ def _construct_single_property_filters(
                             "returning different results each time."
                         )
 
+                # check that the filter returned no entries that had a null or missing value for the filtered property
+                if any(
+                    entry["attributes"].get(prop, entry.get(prop, None)) is None
+                    for entry in reversed_response.get("data", [])
+                ):
+                    raise ResponseError(
+                        f"Filter {reversed_query!r} on field {prop!r} returned entries that had null or missing values for the field."
```

Comment:
I'm confused about this, but maybe it's just because I'm missing the context of where this is put.
If an OPTIMADE filter is supplied on specific values and it's included in the `response_fields` values it _must_ be returned with value `null` if it's not defined. So is this taking care of that as well?

Reply:
Also this and the addition below should be tested, if possible.